helm/flink-kubernetes-operator/templates/rbac.yaml

```diff
@@ -71,13 +71,18 @@ rules:
       - ingresses
     verbs:
       - "*"
+{{/*
+watchNamespaces doesn't need this since the operator handles it
+*/}}
+{{- if not .Values.watchNamespaces }}
   - apiGroups:
       - coordination.k8s.io
     resources:
       - leases
     verbs:
       - "*"
 {{- end }}
```

Comment:
I don't completely understand this. Could we simply remove this entire if logic? It seems that the required role is created later anyways

Reply:
I added the conditional since it doesn't seem like the namespaces in `watchNamespaces` uses leases, but I didn't want the behavior to change for cluster-scoped rules since on line 229 it uses the same rules template.

Unless I've overlooked something when you say that
> It seems that the required role is created later anyways